modules/vactory_page/modules/vactory_page_import/src/Services/PageExportService.php

```diff
@@ -0,0 +1,483 @@
+            if (str_starts_with($key, 'group_')) {
+              foreach ($value as $sub_key => $sub_value) {
+                $field_settings = $widget_settings['fields'][$key][$sub_key];
+                if (isset($field_settings['type'])) {
+                  $normalized_value = $this->normalizeDfData($sub_value, $field_settings);
+                  $group_key = $this->replaceGroup($key);
+                  $node_array['paragraphs'][$paragraph_identifier][$group_key . '|' . $sub_key . '|' . $field_settings['type']] = $normalized_value;
+                }
+              }
+            }
+            else {
+              $field_settings = $widget_settings['fields'][$key];
+              if (isset($field_settings['type'])) {
+                $normalized_value = $this->normalizeDfData($value, $field_settings);
+                $node_array['paragraphs'][$paragraph_identifier][$key . '|' . $field_settings['type']] = $normalized_value;
+              }
+            }
+          }
+        }
+      }
+      if ($paragraph_entity->bundle() == 'vactory_paragraph_multi_template') {
+        $multiple_paragraph_identifier = $paragraph_entity->get('paragraph_key')->value;
+        if (!isset($multiple_paragraph_identifier)) {
+          $type = $paragraph_entity->get('field_multi_paragraph_type')->value;
+          $type = !empty($type) ? $type : 'tab';
+          $paragraph_identifier = "multiple" . $index + 1 . '|' . $type;
+          $paragraph_entity->paragraph_key = $node_id . '|' . $paragraph_identifier;
+          $paragraph_entity->save();
+        }
+        else {
+          $multiple_paragraph_identifier = str_replace($node_id . '|', '', $multiple_paragraph_identifier);
+        }
+        $paragraph_tabs_ids = $paragraph_entity->get('field_vactory_paragraph_tab')->getValue();
+
+        foreach ($paragraph_tabs_ids as $tab_index => $paragraph_tab_id) {
+          $paragraph_tab_entity = Paragraph::load($paragraph_tab_id['target_id']);
+          $tab_title = $paragraph_tab_entity->get('field_vactory_title')->value;
+          $tab_key = $this->toSnakeCase($tab_title);
+          $tab_templates = $paragraph_tab_entity->get('field_tab_templates')->getValue();
+          $paragraph_tab_identifier = $paragraph_tab_entity->get('paragraph_key')->value;
+          if (!isset($paragraph_tab_identifier)) {
+            $paragraph_identifier = "multiple" . $tab_index + 1 . '|' . $tab_key;
+            $paragraph_tab_entity->paragraph_key = $node_id . '|' . $paragraph_identifier;
+            $paragraph_tab_entity->save();
+          }
+          else {
+            $paragraph_tab_identifier = str_replace($node_id . '|' . $multiple_paragraph_identifier . '|', '', $paragraph_tab_identifier);
+          }
+          $normalized_tab_templates = [];
+          foreach ($tab_templates as $tab_template) {
+            $widget_settings = \Drupal::service('vactory_dynamic_field.vactory_provider_manager')->loadSettings($tab_template['widget_id']);
```

Comment:
@ismail-void  [Perf] `\Drupal::service('vactory_dynamic_field.vactory_provider_manager')` move up to `constructNodeArray`